### ISSUE TITLE
include_vars does a hash merge instead of replace (2.0.2.0)

### ISSUE THREAD

**Issue by joonasjussila** [ansible/ansible#15517]
##### ISSUE TYPE

- Bug Report
##### ANSIBLE VERSION

```
ansible 2.0.2.0
  config file = 
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

None, tested on a clean setup
##### OS / ENVIRONMENT

Ubuntu 15.10
##### SUMMARY

Since upgrading to 2.0.2.0, include_vars module is causing hash variables with the same name to be merged instead of being fully replaced.
##### STEPS TO REPRODUCE

playbook.yml:

```
- hosts: localhost
  connection: local
  tasks:
    - include_vars: defaults.yml
    - include_vars: prod.yml
    - debug: var=myvar
```

hosts:

```
localhost
```

defaults.yml:

```
myvar:
  from_defaults:
```

prod.yml:

```
myvar:
  from_prod:
```

Run command:

```
ansible-playbook -i hosts playbook.yml
```
##### EXPECTED RESULTS

Output from 2.0.1.0:

```
PLAY ***************************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [include_vars] ************************************************************
ok: [localhost]

TASK [include_vars] ************************************************************
ok: [localhost]

TASK [debug] *******************************************************************
ok: [localhost] => {
    "myvar": {
        "from_prod": null
    }
}

PLAY RECAP *********************************************************************
localhost                  : ok=4    changed=0    unreachable=0    failed=0   

```
##### ACTUAL RESULTS

Output from 2.0.2.0

```
PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [include_vars] ************************************************************
ok: [localhost]

TASK [include_vars] ************************************************************
ok: [localhost]

TASK [debug] *******************************************************************
ok: [localhost] => {
    "myvar": {
        "from_defaults": null, 
        "from_prod": null
    }
}

PLAY RECAP *********************************************************************
localhost                  : ok=4    changed=0    unreachable=0    failed=0   

```

**Comment by joonasjussila** [comment]
Recreated this in ansible-modules-core: https://github.com/ansible/ansible-modules-core/issues/3508

**Comment by nvigot** [comment]
Could you please reopen this ticket?

**Comment by joonasjussila** [comment]
Reopened to include @nvigot 's pull request, seems that the underlying problem is in the ansible core instead of ansible-modules-core.

**Comment by aalaesar** [comment]
Hello.
I have also reproduced this issue in Ansible 2.1.1.0.
Regards

**Comment by bcoca** [comment]
closing via merge #17486

**Comment by aalaesar** [comment]
Hi
Both tickets are now closed. Shouldn't this one be kept opened ?
Regards.

**Comment by bcoca** [comment]
@aalaesar does the merged patch not fix it for you?

**Comment by aalaesar** [comment]
@bcoca Sorry, I have missed the commit 10f840e so I misunderstood the closing.
Tested it and confirmed it's fixed.
Thank you and @nvigot  for the fix, you're awesome 👍

**Comment by sergeij** [comment]
Will this patch also be merged into the next Ansible v2.1.3 release?
